operator/controllers/manifest_controller_helper_test.go

```diff
@@ -118,32 +118,18 @@ func createImageSpec(name, repo string, ociLayerType OCILayerType) manifestTypes
 	return imageSpec
 }
 
-func NewTestManifest(name string, componentOwner string) *v1alpha1.Manifest {
+func NewTestManifest() *v1alpha1.Manifest {
 	return &v1alpha1.Manifest{
-		TypeMeta: metav1.TypeMeta{
-			Kind: "Manifest",
-		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name + RandString(8),
+			Name:      string(uuid.NewUUID()),
 			Namespace: metav1.NamespaceDefault,
 			Labels: map[string]string{
-				labels.ComponentOwner: componentOwner,
-				labels.CacheKey:       RandString(8),
```

Comment:
from the currently code, basically the `ComponentOwner` will assigned to CacheKey, right? IMO, we should randomize `CacheKey` instead of `ComponentOwner`

```
		if manifestObj.Labels[labels.CacheKey] == "" && manifestObj.Labels[labels.ComponentOwner] != "" {
			// this label ensures caching of manifest rendering clients
			// by the module-manager library
			manifestObj.Labels[labels.CacheKey] = manifestObj.Labels[labels.ComponentOwner]
			labelAdded = true
		}
```
And `ComponentOwner` itself has other purpose if I understood correctly, for example, we need to use it reference cluster kubeConfig secret [here](https://github.com/kyma-project/module-manager/blob/main/operator/internal/pkg/prepare/deploy.go#L116-L134), that's why I create it as a configurable variable for this manifest CR creation so that later can design some testing for it.

And maybe we have some bug here even, the following [code](https://github.com/kyma-project/module-manager/blob/main/operator/internal/pkg/prepare/deploy.go#L116-L123) use componentOwner as manifest processor cache key, it should use `CacheKey` label, right?
```
	kymaOwnerLabel, err := util.GetResourceLabel(manifestObj, labels.ComponentOwner)
	if err != nil {
		return types.ClusterInfo{}, err
	}

	// cluster info record from cluster cache
	kymaNsName := client.ObjectKey{Name: kymaOwnerLabel, Namespace: manifestObj.Namespace}
	processor := processorCache.GetProcessor(kymaNsName)

Reply:
The above mentioned use `ComponentOwner` as  `GetProcessor` cache key leads to inconsistency, because the `SetProcessor` use `CacheKey` as cache key
```
type RendererCache interface {
	// GetProcessor returns the manifest processor cached entry by key passed.
	GetProcessor(key client.ObjectKey) ManifestClient

	// SetProcessor sets the manifest processor cached entry by key passed.
	SetProcessor(key client.ObjectKey, renderSrc ManifestClient)
```